### PR TITLE
[ENG-3854] Add management command to sync all DOI metadata

### PR DESCRIPTION
**Pull request by Johnetordoff**
## QA Notes

Please make verification statements inspired by your code and what your code touches.
- Verify
- Verify

What are the areas of risk?

Any concerns/considerations/questions that development raised?

## Ticket

https://openscience.atlassian.net/browse/ENG-3854

### CODE REVIEW

File: osf/management/commands/sync_doi_metadata.py [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+from framework.celery_tasks import app
+from framework.celery_tasks.handlers import enqueue_task
+
+logger = logging.getLogger(__name__)
+
+
+@app.task(name='osf.management.commands.sync_doi_metadata', max_retries=5, default_retry_delay=60)
+def sync_identifier_doi(identifier):
+    identifier.referent.request_identifier_update('doi')
+    identifier.modified = timezone.now()
```
Comment (mfraezz) [inline osf/management/commands/sync_doi_metadata.py:19]: Minor: `.save` will update this as it's `auto_now=True`; there's no real need to set it.

File: osf/management/commands/sync_doi_metadata.py [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+        )
+        parser.add_argument(
+            '--batch_size',
+            '-b',
+            type=int,
+            default=100,
+            help='number of dois to update in this batch.',
+        )
+        parser.add_argument(
+            '--modified_date',
```
Comment (mfraezz) [inline osf/management/commands/sync_doi_metadata.py:58]: Should this be `required=True`? It's a required positional arg for `sync_doi_metadata` with no default.

File: osf/management/commands/sync_doi_metadata.py [hunk 3]
```diff
@@ -0,0 +1,68 @@
...
+
+from framework.celery_tasks import app
+from framework.celery_tasks.handlers import enqueue_task
+
+logger = logging.getLogger(__name__)
+
+
+@app.task(name='osf.management.commands.sync_doi_metadata', max_retries=5, default_retry_delay=60)
+def sync_identifier_doi(identifier):
+    identifier.referent.request_identifier_update('doi')
```
[the remark this answers is not in the document]
Comment (Johnetordoff) [reply osf/management/commands/sync_doi_metadata.py:18]: @jwalz Thoughts on future-proofing this? is it worth it?

[the remark this answers is not in the document]
Comment (jwalz) [reply osf/management/commands/sync_doi_metadata.py:17]: I'd add object_id__isnull=False to the identifiers query below. It's easy and makes this reusable

**Comment by Johnetordoff** [comment]
@jwalz  @mfraezz I've addressed the comments, but one additional thing I had to do was move the test  under the `api_test` directory, otherwise the conftest.py file under `osf_tests` does not start the celery app for the GHA environment. IT was easier to move the test then to deal with potential test side effects from tampering with the `osf_test` conftest.py. Other then that should be g2g. 

Visibility for Matt these old EZID DOIs should be fixed by this: https://openscience.atlassian.net/browse/ENG-3933

**Comment by jwalz** [comment]
> Visibility for Matt these old EZID DOIs should be fixed by this: https://openscience.atlassian.net/browse/ENG-3933

Do we have confirmation on this? Are we sure the EZID DOIs aren't the 'legacy_doi' dois?

**Comment by Johnetordoff** [comment]
I assume as long as they Identifiers they will grab their refferent's corresponding client and updating the the DOI via that, so the Datacite client will delete the metadata for those, I believe.

File: api_tests/identifiers/managment_commands/test_sync_dois.py [hunk 4]
```diff
@@ -0,0 +1,71 @@
...
+        identifier.modified = timezone.now() - datetime.timedelta(days=1)
+        identifier.save(update_modified=False)
+        return identifier
+
+    @pytest.mark.enable_enqueue_task
+    def test_doi_synced_datacite(self, app, registration, registration_identifier, mock_datacite):
+        assert registration_identifier.modified.date() < datetime.datetime.now().date()
+
+        call_command('sync_doi_metadata', f'-m={datetime.datetime.now()}')
+        assert len(mock_datacite.calls) == 2
```
Comment (jwalz) [inline api_tests/identifiers/managment_commands/test_sync_dois.py:54]: nit: why 2?

File: osf/management/commands/sync_doi_metadata.py [hunk 5]
```diff
@@ -0,0 +1,68 @@
...
+        category='doi',
+        deleted__isnull=True,
+        modified__lte=modified_date,
+        object_id__isnull=False,
+    )[:batch_size]
+    logger.info(f'{"[DRY RUN]: " if dry_run else ""}'
+                f'{identifiers.count()} identifiers to mint')
+
+    for identifier in identifiers:
+        if not dry_run:
```
Comment (jwalz) [inline osf/management/commands/sync_doi_metadata.py:33]: Let's make sure that the referent isn't deleted or withdrawn (or spam? or should deleted catch that?) before we spin up the task.

File: api_tests/identifiers/managment_commands/test_sync_dois.py (hunk 4 above)
Comment (Johnetordoff) [reply api_tests/identifiers/managment_commands/test_sync_dois.py:54]: One call mints the DOI and the other uploads the metadata, the full details are in the client update logic.

File: osf/management/commands/sync_doi_metadata.py (hunk 5 above)
Comment (Johnetordoff) [reply osf/management/commands/sync_doi_metadata.py:33]: The client sends a withdraw request when it has a deleted/private node, I was hoping that would update some of those old EZID DOIs, and potentially other's that might have randomly not been contacted to network issues.

Comment (mfraezz) [reply osf/management/commands/sync_doi_metadata.py:34]: Yep. `not identifier.referent.is_retracted` should probably also be checked below, too.

**Pull request merged**